parquet-thrift/src/main/java/parquet/thrift/struct/ThriftType.java

```diff
@@ -206,6 +237,32 @@ public ThriftField getChildById(short id) {
       }
     }
 
+    /**
+     * Only used for json serialization. Use {@link #isUnion()} instead.
+     *
+     * For backwards compatibility, we don't write the
+     * isUnion field if it's unknown.
+     * Otherwise, it's serialized as a boolean.
+     */
+    @JsonSerialize(include = Inclusion.NON_NULL)
+    @JsonProperty("isUnion")
+    public Boolean jsonIsUnion() {
+      switch (isUnion) {
+        case YES:
+          return true;
+        case NO:
+          return false;
+        case UNKNOWN:
+          return null;
+        default:
+          throw new ParquetRuntimeException("This should never happen"){};
+      }
```

Comment:
possibly move this logic to the IsUnion enum.
public enum IsUnion {
  abstract Boolean toBool();
      YES { Boolean toBool() { return true} },
     NO { Boolean toBool() { return false} },
     UNKNOWN { Boolean toBool() { return null} },
+   }